ios/FluentUI/Vnext/Button/ButtonTokens.swift

```diff
@@ -30,145 +30,291 @@ import UIKit
-class MSFButtonTokens: MSFTokensBase, ObservableObject {
-    @Published public var borderRadius: CGFloat!
-    @Published public var borderSize: CGFloat!
-    @Published public var iconSize: CGFloat!
-    @Published public var interspace: CGFloat!
-    @Published public var padding: CGFloat!
-    @Published public var textFont: UIFont!
-    @Published public var textMinimumHeight: CGFloat!
-    @Published public var textAdditionalHorizontalPadding: CGFloat!
-
-    @Published public var titleColor: UIColor!
-    @Published public var borderColor: UIColor!
-    @Published public var backgroundColor: UIColor!
-    @Published public var iconColor: UIColor!
-
-    @Published public var highlightedTitleColor: UIColor!
-    @Published public var highlightedBorderColor: UIColor!
-    @Published public var highlightedBackgroundColor: UIColor!
-    @Published public var highlightedIconColor: UIColor!
-
-    @Published public var disabledTitleColor: UIColor!
-    @Published public var disabledBorderColor: UIColor!
-    @Published public var disabledBackgroundColor: UIColor!
-    @Published public var disabledIconColor: UIColor!
-
-    @Published public var restShadow1Color: Color!
-    @Published public var restShadow1Blur: CGFloat!
-    @Published public var restShadow1DepthX: CGFloat!
-    @Published public var restShadow1DepthY: CGFloat!
-    @Published public var restShadow2Color: Color!
-    @Published public var restShadow2Blur: CGFloat!
-    @Published public var restShadow2DepthX: CGFloat!
-    @Published public var restShadow2DepthY: CGFloat!
-
-    @Published public var pressedShadow1Color: Color!
-    @Published public var pressedShadow1Blur: CGFloat!
-    @Published public var pressedShadow1DepthX: CGFloat!
-    @Published public var pressedShadow1DepthY: CGFloat!
-    @Published public var pressedShadow2Color: Color!
-    @Published public var pressedShadow2Blur: CGFloat!
-    @Published public var pressedShadow2DepthX: CGFloat!
-    @Published public var pressedShadow2DepthY: CGFloat!
-
-    var style: MSFButtonStyle
-    var size: MSFButtonSize
-
-    init(style: MSFButtonStyle,
-         size: MSFButtonSize) {
+public class ButtonTokens: ControlTokens {
+    public init(style: MSFButtonStyle,
+                size: MSFButtonSize) {
         self.style = style
         self.size = size
 
         super.init()
-
-        self.themeAware = true
-        updateForCurrentTheme()
     }
 
-    @objc open func didChangeAppearanceProxy() {
-        updateForCurrentTheme()
-    }
+    let style: MSFButtonStyle
+    let size: MSFButtonSize
 
-    override func updateForCurrentTheme() {
-        let currentTheme = theme
-        var appearanceProxy: AppearanceProxyType
+    lazy var borderRadius: CGFloat = {
+        switch size {
+        case .small:
+            return globalTokens.borderRadius[.large]
+        case .medium:
+            return globalTokens.borderRadius[.large]
```

Comment:
If they're the same, we should be able do something like
`case .small,
     .medium:`

Reply:
I went back and forth on this one, but I think I'd like to keep them separate still, for a couple of reasons:

* It's easier to skim a large file like this and see, for example, all `.medium` sized values.
* The values aren't intrinsically linked; it's often just coincidence that they happen to match. As such, keeping them as separate instances seems nicer.
* There's a long-term north star goal to have this code generated, and keeping each entry separate will mean simpler generator code.

But that's just an opinion! I'm happy to reconsider if there are cases against separation that I haven't considered.

Reply:
My personal opinion is that having the combined cases makes the code more readable.
I do agree that it creates complexity in the code generator in the future. And I believe we can reconsider if that extra complexity is very costly in the future.
I suspect that it might be a tiny bit less code unless the compiler optimizes it.

Thoughts?

Reply:
Fair enough! I don't feel especially strongly, so I'm happy to fold it together.

Reply:
If/when we try to generate the tokens, I think we could probably group the cases together in some sort of dictionary that holds a list of the cases as the value and the return value as the key? That doesn't seem super complicated, and should let us generate the tokens grouped like this.